src/Services/DataService.test.ts

```diff
@@ -739,32 +836,80 @@ describe("DataService", () => {
   });
 
   it("has an epic for performing a PATCH request", () => {
-    assert.isFunction(fakeService.patchRecordEpic);
+    expect(typeof fakeService.patchRecordEpic).to.equal("function");
   });
 
   describe("patchRecordEpic", () => {
     it("should call patchItem with id and expected result", () => {
       const onSuccess = spy();
       const expectedResult = { id: "123", fullText: "puppy" };
+      const expectedJSONResult = JSON.stringify(expectedResult.fullText);
+      const patchRecordAction = fakeService.actions.patchRecord(expectedResult, { onSuccess });
+      
+      stub(fakeService.serializer, "serialize").returns(expectedJSONResult);
+
+      fakeService.patchRecordEpic(ActionsObservable.of(patchRecordAction), store)
+        .subscribe(noop, noop,
+          () => {
+            expect(mockAdapter.patchItem.calledWithMatch(expectedResult.id, expectedJSONResult)).to.be.true;
+          });
+    });
+
+    it("patchRecordEpic should call normalize before deserialize", () => {
+      const onSuccess = spy();
+      const expectedResult = { id: "123", fullText: "zella puppy normalize" };
       const patchRecordAction = fakeService.actions.patchRecord(expectedResult, { onSuccess });
-      mockAdapter.patchItem.returns(Observable.of(expectedResult));
+      const normalizedStub = stub(fakeService.mapper, "normalize");
 
+      stub(fakeService.serializer, "deserialize").returns(expectedResult);
+      
       fakeService.patchRecordEpic(ActionsObservable.of(patchRecordAction), store)
         .subscribe(noop, noop,
           () => {
-            assert.isTrue(mockAdapter.patchItem.calledWithMatch(expectedResult.id, JSON.stringify({ fullText: "puppy" })));
+            expect(normalizedStub.firstCall.args[0]).to.equal(expectedResult);
+          });
+    });
+
+    it("patchRecordEpic should call transform before serialize", () => {
+      const onSuccess = spy();
+      const expectedResult = { id: "123", fullText: "zella puppy transform" };
+      const patchRecordAction = fakeService.actions.patchRecord(expectedResult, { onSuccess });
+      const transformStub = stub(fakeService.mapper, "transform");
+
+      stub(fakeService.serializer, "serialize").returns(expectedResult);
+      
+      fakeService.patchRecordEpic(ActionsObservable.of(patchRecordAction), store)
+        .subscribe(noop, noop,
+          () => {
+            expect(transformStub.firstCall.args[0]).to.equal(expectedResult);
+          });
+    });
+
+    it("patchRecordItem should serialize the result from transform", () => {
```

Comment:
"patchRecordItem" should be changed to "patchRecordEpic" to be uniform with the other tests.